gemini/reasoning-engine/tutorial_langgraph_cloud_sql_pg_rag_agent.ipynb

```diff
@@ -0,0 +1,1078 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "XtiB5-LVVkv0"
+      },
+      "outputs": [],
+      "source": [
+        "REGION = \"\"  # @param {type:\"string\"}\n",
+        "INSTANCE = \"\"  # @param {type:\"string\"}\n",
+        "DATABASE = \"\"  # @param {type:\"string\"}\n",
+        "MOVIE_TABLE_NAME = \"\"  # @param {type:\"string\"}\n",
+        "BOOK_TABLE_NAME = \"\"  # @param {type:\"string\"}\n",
+        "PASSWORD = input(\"Please provide a password to be used for 'postgres' database user: \")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "dqDjyLpS5zCm"
+      },
+      "source": [
+        "### Grant access to vector store table to IAM users"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "id": "24NjnjF95ySA"
+      },
+      "outputs": [],
+      "source": [
+        "engine = await PostgresEngine.afrom_instance(\n",
+        "    PROJECT_ID, REGION, INSTANCE, DATABASE, user=\"postgres\", password=PASSWORD\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "OaP1LRhPi0y7"
+      },
+      "source": [
+        "### Initialize multiple vector store tables\n",
+        "\n",
+        "The `PostgresEngine` has a helper method `init_vectorstore_table()` that can be used to create a table with the proper schema to store vector embeddings."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 9,
+      "metadata": {
+        "id": "GGd89YWIi2qg"
+      },
+      "outputs": [],
+      "source": [
+        "for table_name in [MOVIE_TABLE_NAME, BOOK_TABLE_NAME]:\n",
+        "    engine = await PostgresEngine.afrom_instance(\n",
+        "        PROJECT_ID, REGION, INSTANCE, DATABASE, user=\"postgres\", password=PASSWORD\n",
+        "    )\n",
+        "\n",
+        "    await engine.ainit_vectorstore_table(\n",
+        "        table_name=table_name,\n",
+        "        vector_size=768,  # Vector size for VertexAI model(textembedding-gecko@latest)\n",
+        "    )"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "sQ1MI8ARi5Rr"
+      },
+      "source": [
+        "### Add embeddings to the vector store\n",
+        "\n",
+        "Load data from a CSV file to generate and insert embeddings to the vector store.\n",
+        "\n",
+        "We will use two datasets:\n",
+        "\n",
+        "* Harry Potter Movie\n",
+        "  - Intro: https://www.kaggle.com/datasets/rounakbanik/the-movies-dataset\n",
+        "  - Data: gs://reasoning-engine-test-1-bucket/harry_potter/harry_potter_movies.json\n",
+        "* Harry Potter Book\n",
+        "  - Intro: https://www.kaggle.com/datasets/shubhammaindola/harry-potter-books\n",
+        "  - Data: gs://reasoning-engine-test-1-bucket/harry_potter/harry_potter_books.json\n"
```

Comment:
We'll need to move these sample data files into the gs://cloud-sample-data or gs://github-repo bucket for consistency and longevity with the rest of the sample data used in this repo.

Reply:
I changed the urls to 
gs://github-repo/generative-ai/gemini/reasoning-engine/harry_potter/harry_potter_movies.json
and 
gs://github-repo/generative-ai/gemini/reasoning-engine/harry_potter/harry_potter_books.json

However, I don't have the access to create file in the gs://github-repo